tests/perforce_utils.py

```diff
@@ -280,7 +281,7 @@ def cleanup(self) -> None:
 
 @pytest.fixture()
 def perforce_workspace(request, perforce_connection, tmpdir):
-    workspace = PerforceWorkspace(perforce_connection, tmpdir)
+    workspace = PerforceWorkspace(perforce_connection.p4, perforce_connection.password, tmpdir)
```

Comment:
Please change the code to pass (and store) the full perforce_connection object, instead of just P4 instance.